app/controllers/wishlists_controller.rb

```diff
@@ -10,16 +10,19 @@ def index
   def new
     @wishlist = Wishlist.new(wishlist_params)
     authorize! :new, @wishlist
+  rescue CanCan::AccessDenied
+    redirect_to wishlists_path, alert: t("generic_error")
   end
 
   def create
     @wishlist = Wishlist.new(wishlist_params)
     authorize! :create, @wishlist
-    if @wishlist.save
-      redirect_to wishlists_path, notice: t(".success")
-    else
-      render :new
-    end
+    @wishlist.save!
+    redirect_to wishlists_path, notice: t(".success")
+  rescue CanCan::AccessDenied
+    redirect_to wishlists_path, alert: t("generic_error")
+  rescue ActiveRecord::RecordInvalid
+    render :new
```

Comment:
Add a message for this as well. We discussed this in Ala's PR, you can see it for reference.